libphobos/m4/druntime/libraries.m4

```diff
@@ -30,3 +30,27 @@ AC_DEFUN([DRUNTIME_LIBRARIES_ZLIB],
     [system_zlib=yes],[system_zlib=no])
   AM_CONDITIONAL([DRUNTIME_ZLIB_SYSTEM], [test "$system_zlib" = yes])
 ])
+
+# DRUNTIME_LIBRARIES_ATOMIC
+# -------------------------
+# Allow specifying whether to use libatomic for
+# 128 bit atomic support.
+AC_DEFUN([DRUNTIME_LIBRARIES_ATOMIC],
+[
+  AC_ARG_WITH(libatomic,
+    AS_HELP_STRING([--without-libatomic],
+                   [Do not use libatomic in core.atomic (default: auto)]))
+
+  DCFG_HAVE_LIBATOMIC=false
+  AS_IF([test "x$with_libatomic" != "xno"], [
+    AC_CHECK_LIB([atomic], [__atomic_load_4], [
+        LIBS="-latomic  $LIBS"
```

Comment:
The `!= xno` is also true when the user did not pass any `--with(out)-libatomic` flags, i.e. this code is executed by default **and** when using `-with-libatomic`. It is only `no` when the user explicitly used `--without-libatomic`.

This idiom is kinda unintuitive, but this seems to be the standard way to do these checks: https://autotools.io/autoconf/arguments.html

So the `AC_CHECK_LIB` check is useful in the default case where the user did not explicitly request or disable `libatomic` to test whether libatomic is actually available. (AFAIK it can be disabled when compiling GCC. I'm not sure if the flag to do this gets passed to our libgphobos configure script as well).